Gemfile

```diff
@@ -1,37 +1,52 @@
-# Variables:
-#
-# SIMP_GEM_SERVERS | a space/comma delimited list of rubygem servers
-# PUPPET_VERSION   | specifies the version of the puppet gem to load
-gem_sources   = ENV.key?('SIMP_GEM_SERVERS') ? ENV['SIMP_GEM_SERVERS'].split(/[, ]+/) : ['https://rubygems.org']
-gem_sources.each { |gem_source| source gem_source }
+gem_sources = ENV.fetch('GEM_SERVERS','https://rubygems.org').split(/[, ]+/)
 
-# mandatory gems
-gem 'coderay'
-gem 'dotenv'
-gem 'metadata-json-lint'
-gem 'parallel'
-gem 'puppet', ENV.fetch('PUPPET_VERSION', '~> 5.5')
-gem 'puppet-lint'
-gem 'puppet-strings'
-gem 'puppetlabs_spec_helper'
-gem 'net-telnet'
-gem 'rake'
-gem 'ruby-progressbar'
-gem 'simp-build-helpers', ENV.fetch('SIMP_BUILD_HELPERS_VERSION', '>= 0.1.0')
-gem 'simp-rake-helpers', ENV.fetch('SIMP_RAKE_HELPERS_VERSION', ['>= 5.11.1', '< 6.0'])
+ENV['PDK_DISABLE_ANALYTICS'] ||= 'true'
 
-group :system_tests do
-  gem 'nokogiri'
-  gem 'beaker', (ENV['SIMP_BEAKER_VERSION']||nil)
-  gem 'beaker-rspec', (ENV['SIMP_BEAKER_RSPEC_VERSION']||nil)
-  gem 'simp-beaker-helpers', ENV.fetch('SIMP_BEAKER_HELPERS_VERSION', ['>= 1.15.1', '< 2.0'])
+gem_sources.each { |gem_source| source gem_source }
+
+group :test do
+  puppet_version = ENV['PUPPET_VERSION'] || '~> 5.5'
+  major_puppet_version = puppet_version.scan(/(\d+)(?:\.|\Z)/).flatten.first.to_i
+  gem 'rake'
+  gem 'puppet', puppet_version
+  gem 'rspec'
+  gem 'rspec-puppet'
+  gem 'hiera-puppet-helper'
+  gem 'puppetlabs_spec_helper'
+  gem 'metadata-json-lint'
+  gem 'puppet-strings'
+  gem 'puppet-lint-empty_string-check',   :require => false
+  gem 'puppet-lint-trailing_comma-check', :require => false
+  gem 'simp-rspec-puppet-facts', ENV['SIMP_RSPEC_PUPPET_FACTS_VERSION'] || '~> 3.1'
+  gem 'simp-rake-helpers', ENV['SIMP_RAKE_HELPERS_VERSION'] || ['> 5.11', '< 6.0']
+  gem 'simp-build-helpers', ENV['SIMP_BUILD_HELPERS_VERSION'] || ['> 0.1', '< 2.0']
+  gem( 'pdk', ENV['PDK_VERSION'] || '~> 1.0', :require => false) if major_puppet_version > 5
 end
 
-# nice-to-have gems (for debugging)
-group :debug do
+group :development do
   gem 'pry'
   gem 'pry-byebug'
   gem 'pry-doc'
 end
 
-#vim: set syntax=ruby:
+group :system_tests do
+  #gem 'beaker'
+  # This is needed until beaker can get it merged and released upstream,
+  # otherwise the acceptance tests will probably fail
+  gem 'beaker', :git => 'https://github.com/trevor-vaughan/beaker', :branch => 'fix_reboot_check_fix'
```

Comment:
As noted in https://github.com/voxpupuli/beaker/pull/1677#discussion_r478729042, the logic of this beaker fix is fundamentally broken (and guts some beaker exceptions in the process)―we shouldn't expect it to merge upstream without extensive revision.  The changes also can't help these acceptance tests pass, unless it's through some kind of oblique side effect (perhaps buying some time in an undiagnosed race condition?).

Reply:
@op-ct I've left comments over there. Presently, it works (regardless of how it works) and unfortunately, I have to push on with other things. I tried all sorts of `sleep` calls first just to see if it was something that was easy to work around, but it wasn't.

Due to time constraints, this is what it is unless someone else has time to make it better.